app/scripts/transaction-manager.js

```diff
@@ -0,0 +1,284 @@
+const EventEmitter = require('events')
+const extend = require('xtend')
+const ethUtil = require('ethereumjs-util')
+const Transaction = require('ethereumjs-tx')
+const BN = ethUtil.BN
+const TxProviderUtil = require('./lib/tx-utils')
+const createId = require('./lib/random-id')
+const normalize = require('./lib/sig-util').normalize
+
+module.exports = class TransactionManager extends EventEmitter {
+  constructor (opts) {
+    super()
+    this.txList = opts.txList || []
+    this._setTxList = opts.setTxList
+    this._unconfTxCbs = {}
+    this.txHistoryLimit = opts.txHistoryLimit
+  // txManager :: tx approvals and rejection cb's
+
+    this.provider = opts.provider
+    this.blockTracker = opts.blockTracker
+    this.txProviderUtils = new TxProviderUtil(this.provider)
+    this.blockTracker.on('block', this.checkForTxInBlock.bind(this))
+    this.getGasMultiplier = opts.getGasMultiplier
+    this.getNetwork = opts.getNetwork
+  }
+
+  getState () {
+    return {
+      transactions: this.getTxList(),
+      unconfTxs: this.getUnapprovedTxList(),
+    }
+  }
+
+//   Returns the tx list
+  getTxList () {
+    return this.txList
+  }
+
+  // Adds a tx to the txlist
+  addTx (txMeta, onTxDoneCb = noop) {
+    var txList = this.getTxList()
+    var txHistoryLimit = this.txHistoryLimit
+    if (txList.length > txHistoryLimit - 1) {
+      var index = txList.findIndex((metaTx) => metaTx.status === 'confirmed' || metaTx.status === 'rejected')
+      index ? txList.splice(index, index) : txList.shift()
+    }
+    txList.push(txMeta)
+    this._saveTxList(txList)
+    // keep the onTxDoneCb around in a listener
+    // for after approval/denial (requires user interaction)
+    // This onTxDoneCb fires completion to the Dapp's write operation.
+    this.once(`${txMeta.id}:signed`, function (txId) {
+      this.removeAllListeners(`${txMeta.id}:rejected`)
+      onTxDoneCb(null, true)
+    })
+    this.once(`${txMeta.id}:rejected`, function (txId) {
+      this.removeAllListeners(`${txMeta.id}:signed`)
+      onTxDoneCb(null, false)
+    })
+
+    this.emit('update')
+    this.emit(`${txMeta.id}:unapproved`, txMeta)
+  }
+
+  // gets tx by Id and returns it
+  getTx (txId, cb) {
+    var txList = this.getTxList()
+    var txMeta = txList.find((txData) => txData.id === txId)
+    return cb ? cb(txMeta) : txMeta
+  }
+
+  //
+  updateTx (txMeta) {
+    var txId = txMeta.id
+    var txList = this.getTxList()
+    var index = txList.findIndex((txData) => txData.id === txId)
```

Comment:
this is getting the index of the tx in the list not the whole tx... Don't actually need the whole tx here just it's index.